cli/command_repository_connect.go

```diff
@@ -79,6 +81,7 @@ func (c *connectOptions) setup(svc appServices, cmd *kingpin.CmdClause) {
 	cmd.Flag("enable-actions", "Allow snapshot actions").BoolVar(&c.connectEnableActions)
 	cmd.Flag("repository-format-cache-duration", "Duration of kopia.repository format blob cache").Hidden().DurationVar(&c.formatBlobCacheDuration)
 	cmd.Flag("disable-repository-format-cache", "Disable caching of kopia.repository format blob").Hidden().BoolVar(&c.disableFormatBlobCache)
+	cmd.Flag("cache-key-derivation-algorithm", "Cache key derivation algorithm").Default(crypto.DefaultKeyDerivationAlgorithm).EnumVar(&c.connectCacheKeyDerivationAlgorithm, crypto.AllowedKeyDerivationAlgorithms()...)
```

Comment:
This flag is specific to clients that connect to the repository via the kopia (API) server.

- Move this flag be moved to the `commandRepositoryConnectServer` struct in `command_repository_connect_server.go`?
- The flag name should also reflect that is specific for this connection type, let's use a more specific and descriptive flag name.
- Update and expand the flag description to clearly describe that this flag is for deriving the (content) cache encryption key for clients of the repository server.
- Add `.Hidden()` for this flag and include "[EXPERIMENTAL]" in the description.
- Is additional validation needed? or does `EnumVar` take care of the validation?